TST: Skip Cython test for editable install

thalassemia opened the pull request:
Skip testing Cython extension for editable installs due to issues identified in https://github.com/scipy/scipy/issues/20540.

thalassemia committed:
TST: Skip Cython test for editable install

[skip circle]

seberg reviewed (approved):
See also https://github.com/cython/cython/issues/5820 for a similar discussion.

This LGTM, I am not 100% sure if there may be a better way to detect "editable", but this also seems fine to just use.

This doesn't remove the pain that you may not be able to develop downstream cython modules if you install NumPy with an editable install.  But there isn't much to do about that...

rgommers committed:
TST: add `IS_EDITABLE` to numpy.testing

Also make it work with meson-python 0.15.0 (that returns an empty list
for `__path__`).

rgommers committed:
TST: pytest.ini: don't pick up pythonapi-compat test in editable mode

rgommers committed:
TST: skip other tests that don't work under editable install

The main reason is that tests with a compile step cannot work,
because headers aren't found.

The distutils test could be made to work, but that isn't worth
the effort since it's deprecated code.

rgommers commented:
Thanks @thalassemia! I tweaked the check for editable install a bit to ensure it works with `meson-python <0.16.0`, and then used it to mark all other failing tests. This was never complete it looks like. Now `pytest numpy` passes.

ngoldbaum commented on pytest.ini:
```diff
@@ -1,6 +1,6 @@
 [pytest]
 addopts = -l
-norecursedirs = doc tools numpy/linalg/lapack_lite numpy/_core/code_generators
+norecursedirs = doc tools numpy/linalg/lapack_lite numpy/_core/code_generators numpy/_core/src/common/pythoncapi-compat
```
oops, thanks for catching that I missed this when I added the submodule

ngoldbaum reviewed (commented):
Should we maybe add a smoketest run with an editable build of numpy?

rgommers commented:
> Should we maybe add a smoketest run with an editable build of numpy?

Yes we definitely should. I'd prefer to open a follow-up issue for that though, since making CI happy that time/iterations.

ngoldbaum commented:
SGTM, merging. Thanks @thalassemia!
